flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js

```diff
@@ -81,38 +109,44 @@ class A32NX_FlightPhaseManager {
         }
     }
 
-    handleFcuInput() {
-        const fcuSelFl = Simplane.getAutoPilotDisplayedAltitudeLockValue("feet") / 100;
-        const fl = Math.round(Simplane.getAltitude() / 100);
-
+    handleFcuAltKnobPushPull() {
         // Try Initiate climb
         if (this.fmc.currentFlightPhase === FmgcFlightPhases.TAKEOFF) {
             this.changeFlightPhase(FmgcFlightPhases.CLIMB);
             return;
         }
 
-        if (this.fmc.currentFlightPhase === FmgcFlightPhases.CLIMB || this.fmc.currentFlightPhase === FmgcFlightPhases.CRUISE) {
-            const dest = this.fmc.flightPlanManager.getDestination();
-            // Try initiate descent
-            if (
-                (!!dest && dest.liveDistanceTo < 200 || !dest) &&
-                fcuSelFl < this.fmc.cruiseFlightLevel
-            ) {
-                this.changeFlightPhase(FmgcFlightPhases.DESCENT);
-                return;
-            }
+        if ((this.fmc.currentFlightPhase === FmgcFlightPhases.CLIMB || this.fmc.currentFlightPhase === FmgcFlightPhases.CRUISE) && canInitiateDes(this.fmc)) {
+            this.changeFlightPhase(FmgcFlightPhases.DESCENT);
+        }
+    }
 
-            // Try initiate early descent
-            if (
-                !!dest && dest.liveDistanceTo >= 200 &&
-                fl > 200 &&
-                fcuSelFl < 200
-            ) {
-                this.changeFlightPhase(FmgcFlightPhases.DESCENT);
+    handleFcuAltKnobTurn() {
+        if (this.fmc.currentFlightPhase === FmgcFlightPhases.CRUISE) {
+            const activeVerticalMode = SimVar.GetSimVarValue('L:A32NX_FMA_VERTICAL_MODE', 'enum');
+            if (canInitiateDes(this.fmc, true) && (activeVerticalMode === 13 || activeVerticalMode === 14 || activeVerticalMode === 15 || activeVerticalMode === 23)) {
+                this.fmc.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.DESCENT);
             }
         }
     }
 
+    handleFcuVSKnob() {
+        if (this.fmc.currentFlightPhase === FmgcFlightPhases.CLIMB || this.fmc.currentFlightPhase === FmgcFlightPhases.CRUISE) {
+            /** a timeout of 100ms is required in order to receive the updated autopilot vertical mode */
+            setTimeout(() => {
```

Comment:
Why is it needed to use a timeout?

Wouldn't it be enough to just do it when the mode has changed? My understanding is that the flight phase gets anyhow updated regularly and not event based.

Reply:
The timeout is needed to give the autopilot the chance to "override" the default action.
Because this happens with a small delay the timeout is needed, however when the AP calls this event it would no longer be needed.